apps/src/blockly/eventHandlers.js

```diff
@@ -28,41 +31,57 @@ export function disableOrphans(blockEvent) {
     blockEvent.element === 'disabled' &&
     !blockEvent.newValue &&
     blockEvent.oldValue;
-
+  if (!blockEvent.workspaceId) {
+    return;
+  }
+  const eventWorkspace = Blockly.Workspace.getById(blockEvent.workspaceId);
   if (
     blockEvent.type === Blockly.Events.BLOCK_MOVE ||
     blockEvent.type === Blockly.Events.BLOCK_CREATE ||
     isEnabledEvent
   ) {
-    if (!blockEvent.workspaceId) {
-      return;
-    }
-    const eventWorkspace = Blockly.Workspace.getById(blockEvent.workspaceId);
     if (!blockEvent.blockId) {
       throw new Error('Encountered a blockEvent without a proper blockId');
     }
     let block = eventWorkspace.getBlockById(blockEvent.blockId);
     if (block) {
-      // Changing blocks as part of this event shouldn't be undoable.
-      const initialUndoFlag = Blockly.Events.getRecordUndo();
-      try {
-        Blockly.Events.setRecordUndo(false);
-        const parent = block.getParent();
-        if (parent && parent.isEnabled()) {
-          const children = block.getDescendants(false);
-          for (let i = 0, child; (child = children[i]); i++) {
-            child.setEnabled(true);
-          }
-        } else if (block.outputConnection || block.previousConnection) {
-          do {
-            block.setEnabled(false);
-            block = block.getNextBlock();
-          } while (block);
-        }
-      } finally {
-        Blockly.Events.setRecordUndo(initialUndoFlag);
+      updateEnabled(block);
+    }
+  } else if (
+    blockEvent.type === Blockly.Events.BLOCK_DRAG &&
```

Comment:
Yes, we can, and it does fire less often. However, this does mean that the blocks stay enabled until the definition block is dropped. Probably not major functional issue with that since a student can't run their code in between or anything. Maybe this is good enough if we think we'll get a fix from Blockly with V10?

![2023-12-07 18-29-55 2023-12-07 18_30_09](https://github.com/code-dot-org/code-dot-org/assets/43474485/e468b95c-a490-41cd-9bee-b448a941fc35)